### DISCUSSION TITLE
Setting the `id` for a catalog seems really hard

### DISCUSSION THREAD

**Discussion by alexgleith** [stac-utils/stac-fastapi#605] [Q&A] [answered]
I have a simple StacAPI custom app.py going, but I can't find an easy way to customise the LandingPage `id`.

Here's what I have so far:

```python
api = StacApi(
    title="Digital Earth Pacific STAC API",
    description="A SpatioTemporal Asset Catalog (STAC) API for Digital Earth Pacific",
    extensions=[ContextExtension()],
    settings=settings,
    client=CoreCrudClient(post_request_model=post_request_model),
    search_get_request_model=get_request_model,
    response_class=ORJSONResponse,
)
```

The PC folks have a huge amount of code customising the client, I think, here: https://github.com/microsoft/planetary-computer-apis/blob/b0471ea9f5e84268294b48bc22432dba93907331/pcstac/pcstac/client.py#L301 and that's where they set it.

Surely it's something everyone wants to change though, so shouldn't it be easier to set?

**Comment by gadomski** [comment, marked as answer]
The `CoreCrudClient` has a `landing_page_id` attribute from the `LandingPageMixin`, so

```python
CoreCrudClient(post_request_model=post_request_model, landing_page_id="my-custom-id")
```

_should_ get you what you want?

**Reply by alexgleith** [reply]
Oh wow, so easy. I did look at that, but must have missed it from the mixin...

Thanks @gadomski! Hope you're well!

**Reply by gadomski** [reply]
You betcha, back atcha!
